src/base/plugins/SUBTREELoader.js

```diff
@@ -16,15 +16,16 @@ function getSubtreeCoordinates( tile, parentTile ) {
 
 	if ( ! parentTile ) {
 
-		return [ 0, 0 ];
+		return [ 0, 0, 0 ];
 
 	}
 
 	const x = 2 * parentTile.__x + ( tile.__subtreeIdx % 2 );
 	const y = 2 * parentTile.__y + ( Math.floor( tile.__subtreeIdx / 2 ) % 2 );
-	//TODO z coord
+	const z = tile.__implicitRoot.implicitTiling.subdivisionScheme === 'OCTREE' ?
+		2 * parentTile.__z + ( Math.floor( tile.__subtreeIdx / 4 ) % 2 ) : 0;
```

Comment:
Interestingly just setting z to "0" here would make all the octtree / quadtree logic "just work" even if we don't have conditions for the two tree types 🤔. Maybe worth making a comment for later

Reply:
I don't think that the `0` value even matters, it could be whatever. In the case of a quadtree, the Z value is not used in any calculus and is called only at the url parsing, but there wont be a "z" coordinate in a quadtree url